examples/bigquery_user_info_updater_tool/cron/bigquery_user_info_updater/tests/test_nested_user_info_updater.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import json
+import os
+import pandas as pd
+
+from google.api_core import exceptions
+from google.cloud import bigquery
+
+from bigquery_user_info_updater.updater_tools import query_creator
+from bigquery_user_info_updater.updater_tools import user_info_updater
+from bigquery_user_info_updater.updater_tools import user_schema
+
+
+class TestNestedUserInfoUpdater(object):
+    """Tests functionality of scripts.UserInfoUpdater.
+
+    Attributes:
+        bq_client(google.cloud.bigquery.client.Client): Client to hold
+            configurations needed for BigQuery API requests.
+        dataset_id(str): ID of the dataset that holds the test table.
+        dataset_ref(google.cloud.bigquery.dataset.DatasetReference): Pointer
+            to the dataset that holds the test table.
+        dataset(google.cloud.bigquery.dataset.Dataset): Dataset that holds the
+            test table.
+        user_info_updates_id(str): The ID of the historical table that holds
+            all rows and updates for all users.
+        temp_user_info_updates_id(str): ID of the intermediary temp table.
+        user_info_final_id(str): The ID of table that holds one up-to-date row
+            per user.
+
+    """
+
+    def setup(self):
```

Comment:
jaketf on Aug 27, 2019 Member
see note on other test about setUp / tearDown,

 @annarudy
annarudy on Oct 17, 2019 Author Member
see note above

Reply:
note is actually below